packages/bitcore-wallet-client/lib/key.js

```diff
@@ -0,0 +1,587 @@
+    c[i] = x[i];
+  });
+  if (c.externalSource) {
+    throw new Error('External Wallets are no longer supported');
+  }
+  c.coin = c.coin || 'btc';
+  c.addressType = c.addressType || Constants.SCRIPT_TYPES.P2SH;
+  c.account = c.account || 0;
+  c.rootPath = c.getRootPath();
+  return {key: k, credentials: c};
+};
+
+
+
+Key.prototype.toObj = function() {
+  var self = this;
+
+  var x = {};
+  _.each(FIELDS, function(k) {
+    x[k] = self[k];
+  });
+  return x;
+};
+
+Key.prototype.isPrivKeyEncrypted = function() {
+  return (!!this.xPrivKeyEncrypted) && !this.xPrivKey;
+};
+
+Key.prototype.get = function(password, getMnemonic) {
+  var keys = {};
+
+  if (this.isPrivKeyEncrypted()) {
+    $.checkArgument(password, 'Private keys are encrypted, a password is needed');
+    try {
+      keys.xPrivKey = sjcl.decrypt(password, this.xPrivKeyEncrypted);
+
+      if (getMnemonic && this.mnemonicEncrypted) {
+        keys.mnemonic = sjcl.decrypt(password, this.mnemonicEncrypted);
+      }
+    } catch (ex) {
+      throw new Error('Could not decrypt');
+    }
+  } else {
+    keys.xPrivKey = this.xPrivKey;
+
+    if (getMnemonic) 
+      keys.mnemonic = this.mnemonic;
+  }
+  return keys;
+};
+
+Key.prototype.encrypt = function(password, opts) {
+  if (this.xPrivKeyEncrypted)
+    throw new Error('Private key already encrypted');
+
+  if (!this.xPrivKey)
+    throw new Error('No private key to encrypt');
+
+  this.xPrivKeyEncrypted = sjcl.encrypt(password, this.xPrivKey, opts);
+  if (!this.xPrivKeyEncrypted)
+    throw new Error('Could not encrypt');
+
+  if (this.mnemonic)
+    this.mnemonicEncrypted = sjcl.encrypt(password, this.mnemonic, opts);
+
+  delete this.xPrivKey;
+  delete this.mnemonic;
+};
+
+Key.prototype.decrypt = function(password) {
+  if (!this.xPrivKeyEncrypted)
+    throw new Error('Private key is not encrypted');
+
+  try {
+    this.xPrivKey = sjcl.decrypt(password, this.xPrivKeyEncrypted);
+    if (this.mnemonicEncrypted) {
+      this.mnemonic = sjcl.decrypt(password, this.mnemonicEncrypted);
+    }
+    delete this.xPrivKeyEncrypted;
+    delete this.mnemonicEncrypted;
+  } catch (ex) {
+    log.error('error decrypting:', ex);
+    throw new Error('Could not decrypt');
+  }
+};
+
+
+Key.prototype.derive = function(password, path) {
+  $.checkArgument(path, 'no path at derive()');
+  var xPrivKey = new Bitcore.HDPrivateKey(this.get(password).xPrivKey,NETWORK);
+  var deriveFn = this.compliantDerivation ? _.bind(xPrivKey.deriveChild, xPrivKey) : _.bind(xPrivKey.deriveNonCompliantChild, xPrivKey);
+  return deriveFn(path);
+};
+
+function _checkCoin(coin) {
+  if (!_.includes(['btc', 'bch'], coin)) throw new Error('Invalid coin');
+};
+
+function _checkNetwork(network) {
+  if (!_.includes(['livenet', 'testnet'], network)) throw new Error('Invalid network');
+};
+
+/*
+ * This is only used on "create"
+ * no need to include/support
+ * BIP45
+ */
+
+Key.prototype.getBaseAddressDerivationPath = function(opts) {
```

Comment:
thanks, will change that.

Reply:
Actually, these are not the same:

`legacyRootPath`  is for getting the path of v1.0 credentials. it will never change, and interpret BIP45, for example.

"getRootPath" is the actual schema, and my change in the future.